Essentials/src/main/java/com/earth2me/essentials/AsyncTimedTeleport.java

```diff
@@ -113,6 +116,26 @@ public void run() {
                     }
                     try {
                         cancelTimer(false);
+
+                        // the target might change location, so check if they can accept teleports.
+                        if (AsyncTeleport.TeleportType.TPA == teleport.getTpType() && timer_teleportTarget instanceof PlayerTarget) {
+                            final Player targetPlayer = Bukkit.getPlayer(((PlayerTarget) timer_teleportTarget).getUUID());
+                            if (targetPlayer == null) return;
```

Comment:
this check comes too late, `getPlayer` could already return null, uuid should always be non null